sbol_utilities/build_planning.py

```diff
@@ -0,0 +1,142 @@
+import sbol3
+import tyto
+
+from sbol_utilities.component import get_subcomponents, get_subcomponents_by_identity
+
+# TODO: Change SBOL_ASSEMBLY_PLAN and sbol3.SBOL_DESIGN to tyto calls after resolution of 
+SBOL_ASSEMBLY_PLAN = 'http://sbols.org/v3#assemblyPlan'
+ASSEMBLY_TYPES = {sbol3.SBOL_DESIGN, SBOL_ASSEMBLY_PLAN}
+
+
+def validate_part_in_backbone(pib: sbol3.Component) -> bool:
+    """Check if a Component represents a part in backbone
+
+    :param plan: Component being validated
+    :return: true if its structure follows the best practices for representing a part in backbone
+    """
+    subcomps = get_subcomponents(pib)
+
+    has_insert = False
+    has_backbone = False
+
+    i = 0
+    while (not has_insert or not has_backbone) and i < len(subcomps):
```

Comment:
I think this can be done more clearly with list comprehensions. That will also help you describe how to handle cases with multiple inserts or multiple backbones.

Reply:
Fixed by 87d053f